dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/FilterJdbcPredicate.java

```diff
@@ -240,6 +248,50 @@ private static String rightOperandSql(QueryOperator operator, Parameter paramete
     return " :" + name;
   }
 
+  @Nonnull
+  private static String multiTextClause(
+      String column, QueryOperator operator, Parameter parameter) {
+    String unnestSql = "unnest(string_to_array(lower(" + column + "), ',')) AS val";
+    String trimmed = "trim(val)";
+    String param = parameter != null ? parameter.name() : null;
+
+    return switch (operator) {
+      case IEQ, EQ, IN ->
+          String.format("exists (select 1 from %s where %s in (:%s))", unnestSql, trimmed, param);
+      case NE, NEQ ->
+          String.format(
+              "not exists (select 1 from %s where %s in (:%s))", unnestSql, trimmed, param);
+      case LIKE, ILIKE ->
+          String.format("exists (select 1 from %s where %s like :%s)", unnestSql, trimmed, param);
+      case NLIKE, NILIKE ->
+          String.format(
+              "not exists (select 1 from %s where %s like :%s)", unnestSql, trimmed, param);
+      case SW ->
+          String.format(
+              "exists (select 1 from %s where %s like CONCAT(:%s, '%%'))",
+              unnestSql, trimmed, param);
+      case EW ->
+          String.format(
+              "exists (select 1 from %s where %s like CONCAT('%%', :%s))",
+              unnestSql, trimmed, param);
+      case NULL ->
+          String.format(
+              "not exists (select 1 from %s where %s is not null and %s <> '')",
+              unnestSql, trimmed, trimmed);
+      case NNULL ->
+          String.format(
+              "exists (select 1 from %s where %s is not null and %s <> '')",
+              unnestSql, trimmed, trimmed);
+      default ->
```

Comment:
Should we remove the validation from dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackerevent/TrackerEventOperationParamsMapper.java and make this return `BadRequestException`?

Reply:
I’ve added the validation in `TrackerEventOperationParamsMapper` since we’re already performing similar validations there. It seemed like the most consistent and appropriate place for this logic. If we decide to keep validation in only one place, I’d suggest removing it from `FilterJdbcPredicate`.

Reply:
I think it would be better to have it in `FilterJdbcPredicate`.
Having it there make sure that the object `FilterJdbcPredicate` is always valid and there is no way to create an invalid object and even more we do not need to replicate this validation.
At the moment there is a missing validation in `SingleEventOperationParamsMapper` about this.